starlette/staticfiles.py

```diff
@@ -158,20 +157,24 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         raise HTTPException(status_code=404)
 
     def lookup_path(
-        self, path: str
-    ) -> typing.Tuple[str, typing.Optional[os.stat_result]]:
+        self, path: Path
+    ) -> typing.Tuple[Path, typing.Optional[os.stat_result]]:
         for directory in self.all_directories:
-            full_path = os.path.realpath(os.path.join(directory, path))
-            directory = os.path.realpath(directory)
-            if os.path.commonprefix([full_path, directory]) != directory:
-                # Don't allow misbehaving clients to break out of the static files
-                # directory.
-                continue
+            original_path = Path(directory).joinpath(path)
+            full_path = original_path.resolve()
+            directory = Path(directory).resolve()
             try:
-                return full_path, os.stat(full_path)
+                stat_result = os.lstat(original_path)
+                full_path.relative_to(directory)
+                return full_path, stat_result
+            except ValueError:
+                # Don't allow misbehaving clients to break out of the static files
+                # directory if not following symlinks.
```

Comment:
Can we remove the double negative here? It took me a long time to understand this.

```suggestion
                # Allow clients to break out of the static files directory if following symlinks.
```

Reply:
Done.